src/Serilog.Extensions.Logging/Extensions/Logging/LoggerProviderCollection.cs

```diff
@@ -36,16 +36,16 @@ public void AddProvider(ILoggerProvider provider)
         {
             if (provider == null) throw new ArgumentNullException(nameof(provider));
 
-            var existing = _providers;
-            var added = existing.Concat(new[] {provider}).ToArray();
+            ILoggerProvider[] existing, added;
 
-#pragma warning disable 420 // ref to a volatile field
-            while (Interlocked.CompareExchange(ref _providers, added, existing) != existing)
-#pragma warning restore 420
+            do
```

Comment:
In this situation, the do-while loop is more appropriate instead of just while loop because it allows you to change variables in one place instead of two.